pkg/kubectl/resource/visitor.go

```diff
@@ -106,9 +113,44 @@ func (i *Info) Visit(fn VisitorFunc) error {
 	return fn(i, nil)
 }
 
+// DynamicClient attempts to make a dynamic client from the Info's RESTClient.
+func (i *Info) DynamicClient() (*dynamic.Client, error) {
+	rc, ok := i.Client.(*restclient.RESTClient)
+	if !ok {
+		return nil, errors.New("could not build dynamic client from RESTClient")
+	}
+
+	dc, err := dynamic.FromRESTClient(rc)
+	if err != nil {
+		return nil, err
+	}
+
+	return dc.ParameterCodec(dynamic.VersionedParameterEncoderWithV1Fallback), nil
+}
+
+// APIResource returns and APIResource for the Info object.
+func (i *Info) APIResource() *unversioned.APIResource {
+	return &unversioned.APIResource{
+		Name:       i.Mapping.Resource,
+		Namespaced: i.Namespaced(),
+		Kind:       i.Mapping.GroupVersionKind.Kind,
+	}
+}
+
 // Get retrieves the object from the Namespace and Name fields
 func (i *Info) Get() (err error) {
-	obj, err := NewHelper(i.Client, i.Mapping).Get(i.Namespace, i.Name, i.Export)
+	var obj runtime.Object
```

Comment:
Same comment here - helper should be able to handle generic objects, and we need to fix helper, not add these ifs